pandas/core/indexes/base.py

```diff
@@ -1392,7 +1396,10 @@ def rename(self, name, inplace=False):
         -------
         new index (of same type and class...etc) [if inplace, returns None]
         """
-        return self.set_names([name], inplace=inplace)
+        if name is not None and not is_hashable(name):
+            raise TypeError('Index.name must be a hashable type')
+        else:
```

Comment:
rather do this in set_names

Reply:
I did this in set_names and a lot of tests failed. Is there a particular reason we can’t keep it here?

Reply:
you are going to need it in set_names as that is the canonical way to do this. that's where it should validate. if we have tests that are clearly in error they should be changed.